zerver/management/commands/deactivate_realm.py

```diff
@@ -9,15 +9,32 @@ class Command(ZulipBaseCommand):
     help = """Script to deactivate a realm."""
 
     def add_arguments(self, parser: ArgumentParser) -> None:
+        parser.add_argument('--redirect_url', metavar='<redirect_url>',
+                            help="URL to which the realm has moved")
         self.add_realm_args(parser, True)
 
     def handle(self, *args: Any, **options: str) -> None:
         realm = self.get_realm(options)
+
         assert realm is not None  # Should be ensured by parser
 
         if realm.deactivated:
             print("The realm", options["realm_id"], "is already deactivated.")
+            if ((not realm.deactivated_redirect) and (options['redirect_url'])):
+                print("Setting the redirect URL to", options['redirect_url'])
+                do_deactivate_realm(realm, redirect_url=options['redirect_url'])
+                print("Done!")
+            elif realm.deactivated_redirect is not None:
+                print("It has been moved to", realm.deactivated_redirect)
             exit(0)
+
         print("Deactivating", options["realm_id"])
-        do_deactivate_realm(realm)
+
+        # Set deactivated_redirect if the redirect_url flag is set and it has
+        # not been set already.
+        if ((not realm.deactivated_redirect) and (options['redirect_url'])):
```

Comment:
Similar to my previous comment, it may be worth allowing changing the redirect URL if it was previously set too.